uniffi_bindgen/src/templates/RustString.rs

```diff
@@ -0,0 +1,41 @@
+// Everybody gets basic string support, since they're such a common data type.
+
+/// This helper receives a borrowed foreign language string and
+/// copies it into an owned rust string. It is naturally tremendously unsafe,
+/// because pointers. The main things to remember as a consumer are:
+///
+///   * You must keep the argument buffer alive for the duration of the call,
+///     and avoid mutating it e.g. from other threads.
+///   * You must eventually free the return value by passing it to the string
+///     destructor defined below. Not just "a string destructor like the one
+///     defined below", the very one defined below for this component!
+///
+/// This function will report a panic via its `ExternError` out arg if given
+/// a null or invalid pointer, or a buffer containing non-utf8 bytes.
+#[no_mangle]
+pub unsafe extern "C" fn {{ ci.ffi_string_alloc_from().name() }}(cstr: *const std::os::raw::c_char, err: &mut uniffi::deps::ffi_support::ExternError) -> *mut std::os::raw::c_char {
+    uniffi::deps::ffi_support::call_with_output(err, || {
```

Comment:
@linacambridge a quick heads-up re: https://github.com/mozilla/uniffi-rs/issues/32, just to note that I manually added panic-handling to this new helper function. We'll probably want to nuance it a bit with whatever more sophisticated thing lands in #32.